### ISSUE TITLE
SCM: Import cleanup

### ISSUE THREAD

**Issue by toddrosner** [rundeck/rundeck#6171]
**Is your feature request related to a problem? Please describe.**
There doesn't appear to be an option for resetting the SCM at the end of a Job run. It appears that SCM is configured once in the Project Settings, and pulled/fetched once.

**Describe the solution you'd like**
A Project Job should have the option to be integrated with the SCM configuration, so that a repository is pulled/fetched on each run if desired.

**Describe alternatives you've considered**
This is a fairly common option amongst other systems to have this capability. While I can write a command to delete the SCM repository at the end of a Job run, I cannot seem to write a command to pull/fetch the repository again. It's not even possible to make use of the same SSH key at the Job level which is configured for SCM.

**Additional context**
In a nutshell, I'm simply looking for a way to clone a Git repo at the beginning of a Job run, then delete it when complete. What are other people doing to handle this?

**Comment by toddrosner** [comment]
I created a workaround for this. 

Instead of using the SCM Plugin, I built an SCM specific Job in the Project. The SCM Job does the work of cleaning the cloned repository, then cloning the repository and setting the branch.

I then created a step in the other Jobs (which require the Git repository); the step is a Job Reference Workflow Step that references the Git specific Job to do the Git work. I passed arguments to the Job Reference as such:

`-Environment ${option.Environment} -Location ${option.Location}`

This in-effect allows me to create one Git clone Job per Project, and then to have the option for all other Jobs in the Project to  reference and use the Git Job. It's obviously possible to create, reference, and use additional Git Jobs per Project as well.

It'd be great if the SCM Plugin had this capability, or if there was a way to reference the SCM Plugin from a Job Step to clean the existing repository, then clone it again.
